code/hsec-tools/src/Security/Advisories.hs

```diff
@@ -417,6 +423,33 @@ advisoryDoc (BRanged _ b) = advisoryDoc b
 advisoryDoc (CodeBlock (T.unpack -> "toml") frontMatter) = pure (frontMatter, mempty)
 advisoryDoc _ = Left "Does not have toml code block as first element"
 
+parseAdvisorySummary :: Block -> Either Text Text
+parseAdvisorySummary = fmap inlineText . firstHeading
+
+firstHeading :: Block -> Either Text Inline
+firstHeading = maybeToEither "Does not have summary heading" . go where
```

Comment:
instead of a bringing a new dependency on *extra* for `maybeToEither`, could you instead do:
```haskell
maybe (Left "Does not have summary hearding") Right . go
```